torchvision/prototype/models/depth/stereo/raft_stereo.py

```diff
@@ -0,0 +1,710 @@
+            context_encoder (MultiLevelContextEncoder): The context encoder. Its input is ``image1``.
+                It has multi-level output and each level will have 2 parts:
+
+                - one part will be used as the actual "context", passed to the recurrent unit of the ``update_block``
+                - one part will be used to initialize the hidden state of the of the recurrent unit of
+                  the ``update_block``
+
+            corr_pyramid (CorrPyramid1d): Module to buid the correlation pyramid from feature encoder output
+            corr_block (CorrBlock1d): The correlation block, which uses the correlation pyramid indexes
+                to create correlation features. It takes the coordinate of the centroid pixel and correlation pyramid
+                as input and returns the correlation features.
+                It must expose an ``out_channels`` attribute.
+
+            update_block (MultiLevelUpdateBlock): The update block, which contains the motion encoder, and the recurrent unit.
+                It takes as input the hidden state of its recurrent unit, the context, the correlation
+                features, and the current predicted depth. It outputs an updated hidden state
+            depth_head (nn.Module): The depth head block will convert from the hidden state into changes in depth.
+            mask_predictor (nn.Module, optional): Predicts the mask that will be used to upsample the predicted flow.
+                If ``None`` (default), the flow is upsampled using interpolation.
+            slow_fast (bool): A boolean that specify whether we should use slow-fast GRU or not. See RAFT-Stereo paper
+                on section 3.4 for more detail.
+            num_iters (int): The default number of iteration update during forward pass. Take note that this will only
+                be used if num_iters is not specified on the forward call.
+        """
+        super().__init__()
+        _log_api_usage_once(self)
+
+        self.feature_encoder = feature_encoder
+        self.context_encoder = context_encoder
+
+        self.base_downsampling_ratio = feature_encoder.base_downsampling_ratio
+        self.num_level = self.context_encoder.num_level
+        self.corr_pyramid = corr_pyramid
+        self.corr_block = corr_block
+        self.update_block = update_block
+        self.depth_head = depth_head
+        self.mask_predictor = mask_predictor
+
+        hidden_dims = self.update_block.hidden_dims
+        # Follow the original implementation to do pre convolution on the context
+        # See: https://github.com/princeton-vl/RAFT-Stereo/blob/main/core/raft_stereo.py#L32
+        self.context_convs = nn.ModuleList(
+            [nn.Conv2d(hidden_dims[i], hidden_dims[i] * 3, kernel_size=3, padding=1) for i in range(self.num_level)]
+        )
+        self.slow_fast = slow_fast
+        self.num_iters = num_iters
+
+    def forward(self, image1: Tensor, image2: Tensor) -> List[Tensor]:
```

Comment:
Previously we have `num_iters` as input to forward as this is also the case in the [original implementation](https://github.com/princeton-vl/RAFT-Stereo/blob/main/core/raft_stereo.py#L70). However, this violate the fx constraint, and we would need to wrap the big for loop (line 423 - 463) in order to make the model fx compatible.

Currently I decide to set the num_iters in the constructor instead, and users still can modify during inference by set it before calling the forward. However I am still not too sure this is a good idea, do you have any suggestion @datumbox ?